main/lsp/lsp_helpers.cc

```diff
@@ -280,6 +280,41 @@ optional<string> findDocumentation(string_view sourceCode, int beginIndex) {
             // Do nothing for a one-line sig block
         }
 
+        // Handle multi-line sig block
+        else if (absl::StartsWith(line, "}")) {
+            // ASSUMPTION: We either hit the start of file or a `sig {`.
+            //
+            // Note that this will not properly handle brace blocks where the
+            // closing brace is not on its own line, such as
+            // ```
+            // sig { params(foo: Foo)
+            //       .returns(Bar) }
+            // ```
+            it++;
+            while (
+                // SOF
+                it != all_lines.rend()
+                // Start of sig block
+                && !(absl::StartsWith(absl::StripAsciiWhitespace(*it), "sig {") ||
```

Comment:
What do you think about factoring the code so that we only have to strip whitespace once, instead of possibly 5 times per loop?